src/ride/track.c

```diff
@@ -4880,15 +4881,491 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+    uint16 z = (uint16) (edi & 0xFFFF); // di
+
+    // push    eax
+    // push    ebx
+    // push    ecx
+    // push    edx
+    // push    edi
+
+    // and     ax, 0FFE0h
+    uint16 temp_ax = (uint16) floor2(x, 32);
+
+    // and     cx, 0FFE0h
+    uint16 temp_cx = (uint16) floor2(y, 32);
+
+    // mov     bl, dl
+    uint8 bl = rideIndex;
+    // mov     dx, di
+    uint16 temp_dx = z;
+    // shr     dx, 3
+    temp_dx = temp_dx >> 3;
+
+    // movzx   esi, cx
+    uint32 temp_esi = y;
+
+    // shl     esi, 8
+    temp_esi = temp_esi << 8;
+
+    // or      si, ax
+    temp_esi = (temp_esi & 0xFFFF0000) + (temp_esi & temp_ax);
+
+    // shr     esi, 3
+    temp_esi = temp_esi >> 3;
+
+    // mov     esi, tile_map_element_pointers[esi]
+    rct_map_element *temp_esi_map = TILE_MAP_ELEMENT_POINTER(temp_esi);
+
+
+    while (true) {
+        // ---- loc_6CD9F0:
+        // mov     al, [esi]
+        uint8 temp_al = temp_esi_map->type;
+
+        // and     al, 3Ch
+        temp_al = temp_al & 0x3C;
+
+        // cmp     al, 8
+        // jnz     short loc_6CDA0C
+        if (temp_al == 8) {
+            // cmp     dl, [esi+2]
+            // jnz     short loc_6CDA0C
+            if ((temp_dx & 0xff) == temp_esi_map->base_height) {
+                // cmp     bl, [esi+7]
+                // jnz     short loc_6CDA0C
+                if (rideIndex == temp_esi_map->properties.surface.ownership) {
+                    // cmp     byte ptr [esi+4], 65h ; 'e'
+                    // jz      loc_6CDB35
+                    if (temp_esi_map->properties.surface.slope == 0x65) {
+                        return loc_6CDB35();
+                    }
+                }
+            }
+        }
+
+        // ---- loc_6CDA0C:
+        // add     esi, 8
+        // test    byte ptr [esi-7], 80h
+        // jz      short loc_6CD9F0
+        temp_esi += 8;
+        if (RCT2_GLOBAL(temp_esi - 7, uint8) != 0x80) {
+            break;
+        }
+    }
+
+
+    // mov     edx, [esp+14h+var_10]
+    uint32 temp_edx = RCT2_GLOBAL(esp + 0x14 + 0x10, uint32);
+
+    // or      dh, dh
+    // jnz     loc_6CDE43
+    if ((temp_edx & 0xFF00) != 0) {
+        return loc_6CDE43();
+    }
+
+    // mov     eax, [esp+14h+var_4]
+    eax = eax;
+
+    // mov     ecx, [esp+14h+var_C]
+    ecx = ecx;
+
+    // and     ax, 0FFE0h
+    uint16 x_floored = (uint16) floor2(x, 32);
+
+    // and     cx, 0FFE0h
+    uint16 y_floored = (uint16) floor2(y, 32);
+
+    // mov     edx, [esp+14h+var_14]
+    // shr     dx, 3
+    uint16 tmp_dx = z >> 3;
+
+    // mov     dh, dl
+    tmp_dx = (tmp_dx & 0xFF) + ((tmp_dx & 0xFF) << 8);
+
+    // add     dh, 4
+    tmp_dx += 0x4 << 8;
+
+    // mov     bl, 0Fh
+    // call    can_construct_at
+    // TODO: find out what these somethings are
+    // jb      loc_6CDE4C
+    if (!map_can_construct_at(x_floored, y_floored, something, something_else, 0x0F)) {
+        // TODO: should this be inverted?
```

Comment:
Zlow and zhigh are normally the base height and the clearance height of your new map element. If you look just above dl and dh look suspiciously like they have a z variable. Z coordinates come in two different forms: the actual physical coordinate; and the map element z coordinate (z/8).